### PR TITLE
Feat/smtp email support for email service

### PR DESCRIPTION
**Pull request by Ellenn-A**
# Pull Request

## Checklist
- [x] Have you read Digital Catapult's [Code of Conduct](https://github.com/digicatapult/.github/blob/main/CODE_OF_CONDUCT.md)?
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] New and existing unit tests pass locally with my changes.

## PR Type

Please delete options that are irrelevant.

- [ ] Bug Fix
- [x] Feature
- [x] Documentation Update
- [ ] Code style update (formatting, local variables)
- [ ] Breaking Change (fix or feature that would cause existing functionality to change)

## Linked tickets

https://digicatapult.atlassian.net/jira/software/c/projects/ENG/boards/206?selectedIssue=VR-103&sprints=715
## High level description

Added SMTP Communication Email support to the email service.

## Detailed description

Used [smtp4dev](https://github.com/rnwood/smtp4dev) to bring up a test email server to test email sending. Added `SMTP_EMAIL` as an option to `EMAIL_TRANSPORT` env. By default we are still using `STREAM`, if you do change this to  `SMTP_EMAIL` your emails will no longer be printed to the console and will be instead sent to the test server this can be found on `http://localhost:5000/`. 

## Describe alternatives you've considered

N/A

## Operational impact

N/A

## Additional context
N/A

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -99,6 +99,10 @@ services:
       - ISSUANCE_SCHEMA_POLICY=EXISTING_OR_NEW
       - ISSUANCE_CRED_DEF_POLICY=EXISTING_OR_NEW
       - DEMO_MODE=true
+      - SMTP_HOST=localhost
+      - SMTP_PASS=
+      - SMTP_PORT=2525
+      - SMTP_USER=
```
Comment (mattdean-digicatapult) [inline docker-compose.yml:105]: Let's keep the `smtp` connection internal to the docker network if we can. Cahnge `SMTP_HOST` to `smtp4dev` and `SMTP_PORT` to `25`.

File: test/integration/smtpEmail.test.ts [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+describe('SMTP email', () => {
+  let server: { app: express.Express; cloudagentEvents: VeritableCloudagentEvents }
+
+  withCompanyHouseMock()
+
+  describe('create invitation and check it has been registered in the email server (happy path)', function () {
+    beforeEach(async () => {
+      process.env.EMAIL_TRANSPORT = 'SMTP_EMAIL'
+      resetContainer()
+      container.registerInstance(Env, new Env())
```
Comment (mattdean-digicatapult) [inline test/integration/smtpEmail.test.ts:45]: I really don't like this hack (which I advised you to create). There's a couple of things I'd like you to try:

1. if you remove `container.registerInstance(Env, new Env())` do the tests still pass?
2. now rename the string of the top level describe of any other test file in the integration tests to start with a letter bigger than `S`. I want to change the order some tests run to make sure the cleanup we're doing here actually works
3. If both of these two are fine move the logic of overriding the env and reseting the contianer to a helper that executes the `beforeEach` and `afterEach`. That I think will be clean enough.
4. If either 1 or 2 don't pass we should probably jump on a call

**Review by mattdean-digicatapult — CHANGES_REQUESTED** [review]

File: docker-compose.yml [hunk 3]
```diff
@@ -136,6 +140,14 @@ services:
       - POSTGRES_HOST=postgres-veritable-cloudagent-bob
       - WALLET_ID=bob
       - WALLET_KEY=bob-key
+  # -------------------- email -------------------------------#
+  smtp4dev:
+    image: rnwood/smtp4dev
+    container_name: smtp4dev
+    ports:
+      - '5000:80' # Port for accessing smtp4dev's web interface
+      - '2525:25' # Port for SMTP connections
```
[the remark this answers is not in the document]
Comment (Ellenn-A) [reply docker-compose.yml:149]: this doesn't work when i remove this line `- '2525:25' # Port for SMTP connections`

[the remark this answers is not in the document]
Comment (mattdean-digicatapult) [reply docker-compose.yml:149]: why?

[the remark this answers is not in the document]
Comment (mattdean-digicatapult) [reply docker-compose.yml:149]: I think it's because alice is running locally and therefore sending the emails. This is fine in that case

**Review by mattdean-digicatapult — APPROVED** [review]

**Review by mattdean-digicatapult — APPROVED** [review]

**Pull request merged**